### PR TITLE
feat(components): Toast Component

### PR DESCRIPTION
**Pull request by eddysims**
## Motivations

We are moving towards a better way to display messages to the user. Toasts are the way of the future.

### Added

- Toast Component

### Changed

- 

### Deprecated

- 

### Removed

- 

### Fixed

- 

### Security

- 

## Testing

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

File: packages/components/src/Icon/Icon.tsx [hunk 1]
```diff
@@ -45,6 +45,7 @@ export function Icon({ name, color, customColor, size = "base" }: IconProps) {
       xmlns="http://www.w3.org/2000/svg"
       viewBox={viewBox}
       className={svgClassNames}
+      name={name}
```
Comment (seanhealy) [inline packages/components/src/Icon/Icon.tsx:48]: Why did you add name here?

File: packages/components/src/Toast/Toast.css [hunk 2]
```diff
@@ -0,0 +1,50 @@
+:root {
+  --toast-shadow: 0 var(--space-base) var(--space-base) rgba(0, 0, 0, 0.16),
```
Comment (seanhealy) [inline packages/components/src/Toast/Toast.css:2]: Are we able to use `--shadow-base` from here here?

packages/design/src/shadows.css

File: packages/components/src/Toast/Toast.css [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+.container {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  position: fixed;
+  bottom: var(--space-large);
+  left: 50%;
+  z-index: 10000;
```
Comment (seanhealy) [inline packages/components/src/Toast/Toast.css:13]: Would probably be good to use or add one of these:

packages/design/src/Elevations.mdx

File: packages/components/src/Toast/Toast.mdx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+- Don't use `Dismiss` or `Cancel` as an action label.
+  - Examples of action labels: `Undo`, `View`, `Refresh`.
+
+## Props
+
+<Props of={Toast} />
+
+## Variations
+
+Toasts have 3 variations, `default`, `success`, and `error`. Each should be used
```
Comment (seanhealy) [inline packages/components/src/Toast/Toast.mdx:59]: How do you feel about `info` instead of `default`?

File: tsconfig.base.json [hunk 5]
```diff
@@ -21,7 +21,7 @@
     // "removeComments": true,                /* Do not emit comments to output. */
     // "noEmit": true,                        /* Do not emit outputs. */
     // "importHelpers": true,                 /* Import emit helpers from 'tslib'. */
-    // "downlevelIteration": true,            /* Provide full support for iterables in 'for-of', spread, and destructuring when targeting 'ES5' or 'ES3'. */
+    "downlevelIteration": true,            /* Provide full support for iterables in 'for-of', spread, and destructuring when targeting 'ES5' or 'ES3'. */
```
Comment (seanhealy) [inline tsconfig.base.json:24]: No problem with adding this at all. I'm just curious what required it for learning purposes.

File: packages/components/src/Toast/Toast.tsx [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+  }
+
+  function getTimeout() {
+    const min = 2000;
+    const max = 5000;
+    const time = message.length * 100;
+
+    if (time < min) return min;
+    if (time > max) return max;
+    return time;
```
Comment (seanhealy) [inline packages/components/src/Toast/Toast.tsx:110]: I wonder if `Math.min` and `Math.max` would be better here?

File: packages/components/src/Toast/Toast.tsx [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+          onMouseLeave={() => startTimer()}
+          exit={{
+            opacity: 0,
+            scale: 0.8,
+            height: 0,
+            transition: { duration: 0.4 },
+          }}
+        >
+          <div className={styles.slice}>
+            {icon && (
```
Comment (seanhealy) [inline packages/components/src/Toast/Toast.tsx:60]: There should always be an icon now no?

File: packages/components/src/Toast/Toast.tsx [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+              <div className={styles.icon}>
+                <Icon color={icon.color} name={icon.name} />
+              </div>
+            )}
+
+            <div className={styles.message}>{message}</div>
```
Comment (seanhealy) [inline packages/components/src/Toast/Toast.tsx:66]: Would probably be better to use `Text` or `Typography` here.

File: packages/components/src/Toast/showToast.tsx [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+  const toastRef = useRef() as MutableRefObject<ToastRef>;
+  useEffect(() => toastRef.current.add(props));
+  return <ToastContainer ref={toastRef} />;
+}
+
+/**
+ * Ignoring no-explicit-any as the ToastInternal takes no props, however
+ * since it is used in a forwardRef, it needs the second param.
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (seanhealy) [inline packages/components/src/Toast/showToast.tsx:43]: `unknown` might be better in this case.

File: packages/components/src/Icon/Icon.tsx (hunk 1 above)
Comment (eddysims) [reply packages/components/src/Icon/Icon.tsx:48]: going to change this `data-testid`

File: packages/components/src/Toast/Toast.mdx (hunk 4 above)
Comment (eddysims) [reply packages/components/src/Toast/Toast.mdx:59]: lets change `default` to `info` and default to `info`

File: packages/components/src/Toast/Toast.tsx (hunk 6 above)
Comment (eddysims) [reply packages/components/src/Toast/Toast.tsx:111]: @seanhealy I don't think that there is anything to gain from changing this. I think that the readability of what is here is more simple.

**Comment by eddysims** [comment]
@seanhealy comments addressed. back to you!
